src/corefx/System.Globalization.Native/pal_collation.c

```diff
@@ -588,47 +592,164 @@ int32_t GlobalizationNative_IndexOfOrdinalIgnoreCase(
 }
 
 /*
- Return value is a "Win32 BOOL" (1 = true, 0 = false)
- */
-int32_t GlobalizationNative_StartsWith(
-                        SortHandle* pSortHandle,
-                        const UChar* lpTarget,
-                        int32_t cwTargetLength,
-                        const UChar* lpSource,
-                        int32_t cwSourceLength,
-                        int32_t options)
+ collation element is an int used for sorting. It consists of 3 components:
+    * primary - first 16 bits, representing the base letter
+    * secondary - next 8 bits, typically an accent
+    * tertiary - last 8 bits, typically the case
+
+An example (the numbers are made up to keep it simple)
+    a: 1 0 0
+    ą: 1 1 0
+    A: 1 0 1
+    Ą: 1 1 1
+
+    this method returns a mask that allows for characters comparison using specified Collator Strength
+*/
+static int32_t GetCollationElementMask(UColAttributeValue strength)
 {
-    int32_t result = FALSE;
-    UErrorCode err = U_ZERO_ERROR;
-    const UCollator* pColl = GetCollatorFromSortHandle(pSortHandle, options, &err);
+    switch (strength)
+    {
+        case UCOL_PRIMARY:
+            return UCOL_PRIMARYORDERMASK;
```

Comment:
do we need to assert on the primary strength? as I mentioned in other comment we'll not expect this strength in the simple case.

Reply:
Personally, I am not a big fan of asserts, I prefer to add a lot of unit tests. If we ever set the strength to `UCOL_PRIMARY` all the tests are going to fail so I don't think that we need it.